frontend/packages/metal3-plugin/src/components/dashboard/utils.ts

```diff
@@ -0,0 +1,57 @@
+import { StatusGroupMapper } from '@console/internal/components/dashboard/inventory-card/inventory-item';
+import { InventoryStatusGroup } from '@console/internal/components/dashboard/inventory-card/status-group';
+
+import { getSimpleHostStatus } from '../../utils/host-status';
+import {
+  HOST_ERROR_STATES,
+  HOST_PROGRESS_STATES,
+  HOST_PROVISIONING_STATES,
+  HOST_REGISTERING_STATES,
+  HOST_SUCCESS_STATES,
+} from '../../constants';
+
+const BMH_STATUS_GROUP_MAPPER = {
+  [InventoryStatusGroup.OK]: [...HOST_SUCCESS_STATES],
+  [InventoryStatusGroup.PROGRESS]: [
+    ...HOST_PROGRESS_STATES,
+    ...HOST_PROVISIONING_STATES,
+    ...HOST_REGISTERING_STATES,
+  ],
+  [InventoryStatusGroup.ERROR]: [...HOST_ERROR_STATES],
+};
+
+export const getBMHStatusGroups: StatusGroupMapper = (bmhs) => {
+  const groups = {
+    [InventoryStatusGroup.OK]: {
+      statusIDs: ['ready', 'provisioned'],
+      count: 0,
+      filterType: 'host-status',
+    },
+    [InventoryStatusGroup.ERROR]: {
+      statusIDs: ['error'],
+      count: 0,
+      filterType: 'host-status',
+    },
+    [InventoryStatusGroup.PROGRESS]: {
+      statusIDs: ['registering', 'provisioning'],
+      count: 0,
+      filterType: 'host-status',
+    },
+    [InventoryStatusGroup.NOT_MAPPED]: {
+      statusIDs: ['other'],
+      count: 0,
+      filterType: 'host-status',
+    },
+  };
+
+  bmhs.forEach((bmh) => {
+    const status = getSimpleHostStatus(bmh);
```

Comment:
This function does not exist any more. `getBMHStatusGroups` will now need to take `bmh` and `nodeMaintenance` to correctly identify host status. See https://github.com/openshift/console/blob/master/frontend/packages/metal3-plugin/src/utils/host-status.ts#L57

Reply:
function updated. Im fetching bmhs, nodes, hosts and node maintenances now